From c4d78b8bc05d44b8a46b5d2db6b52b42a101ef2d Mon Sep 17 00:00:00 2001
From: Thorsten Glaser <tg@mirbsd.de>
Date: Mon, 24 Aug 2015 10:53:00 -0400
Subject: [PATCH] FindJNI: Add support for x32 architecture on Linux (#15710)

This architecture uses an amd64 kernel but x32 userland.
Use CMAKE_LIBRARY_ARCHITECTURE to distinguish it for now.
---
 Modules/FindJNI.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Modules/FindJNI.cmake b/Modules/FindJNI.cmake
index d248fe1..d31fea3 100644
--- a/Modules/FindJNI.cmake
+++ b/Modules/FindJNI.cmake
@@ -42,7 +42,11 @@ macro(java_append_library_directories _var)
     # 1.6.0_18 + icedtea patches. However, it would be much better to base the
     # guess on the first part of the GNU config.guess platform triplet.
     if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+      if(CMAKE_LIBRARY_ARCHITECTURE STREQUAL "x86_64-linux-gnux32")
+        set(_java_libarch "x32" "amd64" "i386")
+      else()
         set(_java_libarch "amd64" "i386")
+      endif()
     elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^i.86$")
         set(_java_libarch "i386")
     elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^alpha")
-- 
2.5.0

